src/spline_agent/exceptions.py

```diff
@@ -12,16 +12,29 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-class LineageTrackingContextNotInitialized(BaseException):
-    """ Lineage harvesting context was not properly initialized"""
+class LineageTrackingContextNotInitializedError(BaseException):
```

Comment:
deriving from BaseException is not best practice; it's even mentiond in the documentation: https://stackoverflow.com/questions/60745841/inheriting-from-baseexception-vs-exception#:~:text=Note%20that%20while%20the%20default,s%20thread%20equivalent%2C%20the%20Task.

Rather, derive from `Exception`, as explained in the SO post and/or the PEP8 guide: https://peps.python.org/pep-0008/#programming-recommendations

Reply:
Nice one! My Java intuition suggested me that it's the `Exception` that must be root of the hierarchy with the `BaseException` being one of its subclasses. It turned out to be vice versa in Python 😲 

Fixed.